test/CMakeLists.txt

```diff
@@ -99,6 +119,18 @@ macro(_add_test _NAME _DEPENDS)
     add_dependencies(${_NAME} ${_NAME}_data)
     add_test(NAME ${_NAME} COMMAND ${_NAME})
     target_link_libraries(${_NAME} fdeep Threads::Threads doctest::doctest)
+    # GNU in linux doesn't support "-mbig-obj" option.
+    if(CMAKE_COMPILER_IS_GNUCXX)
+        include(CheckCXXCompilerFlag)
+        check_cxx_compiler_flag("-Wa,-mbig-obj" GNU_BIG_OBJ_FLAG_ENABLE)
```

Comment:
If GNU support `-Wa,-mbig-obj`, `GNU_BIG_OBJ_FLAG_ENABLE` will be set `1`. I test that on Windows `GNU_BIG_OBJ_FLAG_ENABLE=1` and on Centos 9 `GNU_BIG_OBJ_FLAG_ENABLE` is empty, so that `-Wa,-mbig-obj` is support on Windows but not on linux.